packages/widgets/src/lib/components/InferenceWidget/shared/WidgetOuputConvoBubble/WidgetOutputConvoBubble.svelte

```diff
@@ -1,14 +1,64 @@
 <script lang="ts">
+	import type { SpecialTokensMap } from "@huggingface/tasks";
+	import { marked, type MarkedOptions } from "marked";
+
 	export let position: "left" | "right" | "center";
 	export let text: string;
+	export let specialTokensMap: SpecialTokensMap | undefined = undefined;
+
+	$: tokens = marked.lexer(sanitizeMd(text));
 
+	const PUBLIC_SEP_TOKEN = "</s>";
 	const CLASSES: Record<typeof position, string> = {
 		left: "rounded-2xl mr-7 place-self-start bg-gray-50 dark:bg-gray-850 dark:text-gray-200",
 		right: "rounded-2xl ml-7 bg-blue-600 text-white",
 		center: "self-center justify-center border-b text-center pb-4",
 	};
+
+	function sanitizeMd(md: string) {
+		let ret = md
+			.replace(/<\|[a-z]*$/, "")
+			.replace(/<\|[a-z]+\|$/, "")
+			.replace(/<$/, "")
+			.replaceAll(PUBLIC_SEP_TOKEN, " ")
+			.replaceAll(/<\|[a-z]+\|>/g, " ")
+			.replaceAll(/<br\s?\/?>/gi, "\n")
+			.replaceAll("<", "&lt;")
+			.trim();
+
+		const stopTokens = ["<|endoftext|>"];
+		if (typeof specialTokensMap?.eos_token === "string") {
+			stopTokens.push(specialTokensMap.eos_token);
+		}
+		for (const stop of stopTokens) {
+			if (ret.endsWith(stop)) {
+				ret = ret.slice(0, -stop.length).trim();
+			}
+		}
```

Comment:
do I need these lines considering that we already gave https://github.com/huggingface/huggingface.js/blob/59d641b6a7ccdfdf86596f69e4845ef2680e3834/packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte#L140-L144

cc: @xenova

Reply:
I think it is, because this just controls stripping of the final special token (output), right? (and is different to the rendered chat template / input).